substrate/frame/support/src/traits/tokens/asset_ops/common_strategies.rs

```diff
@@ -0,0 +1,459 @@
+	fn default() -> Self {
+		Self(())
+	}
+}
+impl<Flavor> InspectStrategy for CanUpdate<Flavor> {
+	type Value = bool;
+}
+impl<Flavor> UpdateStrategy for CanUpdate<Flavor> {
+	type Update<'u> = bool;
+	type Success = ();
+}
+
+/// This trait converts the given [UpdateStrategy] into the corresponding [CanUpdate] strategy
+/// representing the ability to update the asset using the provided strategy.
+pub trait AsCanUpdate: Sized + UpdateStrategy {
+	fn as_can_update(self) -> CanUpdate<Self>;
+}
+impl<T: UpdateStrategy> AsCanUpdate for T {
+	fn as_can_update(self) -> CanUpdate<Self> {
+		CanUpdate(self)
+	}
+}
+
+/// The `AutoId` is an ID assignment approach intended to be used in
+/// [`"create" strategies`](CreateStrategy).
+///
+/// It accepts the `Id` type of the asset.
+/// The "create" strategy should report the value of type `ReportedId` upon successful asset
+/// creation.
+pub type AutoId<ReportedId> = DeriveAndReportId<(), ReportedId>;
+
+/// The `PredefinedId` is an ID assignment approach intended to be used in
+/// [`"create" strategies`](CreateStrategy).
+///
+/// It accepts the `Id` that should be assigned to the newly created asset.
+///
+/// The "create" strategy should report the `Id` value upon successful asset creation.
+pub type PredefinedId<Id> = DeriveAndReportId<Id, Id>;
+
+/// The `DeriveAndReportId` is an ID assignment approach intended to be used in
+/// [`"create" strategies`](CreateStrategy).
+///
+/// It accepts the `Params` and the `Id`.
+/// The `ReportedId` value should be computed by the "create" strategy using the `Params` value.
+///
+/// The "create" strategy should report the `ReportedId` value upon successful asset creation.
+///
+/// An example of ID derivation is the creation of an NFT inside a collection using the
+/// collection ID as `Params`. The `ReportedId` in this case is the full ID of the NFT.
+#[derive(RuntimeDebug, PartialEq, Eq, Clone, Encode, Decode, MaxEncodedLen, TypeInfo)]
+pub struct DeriveAndReportId<Params, ReportedId> {
+	pub params: Params,
+	_phantom: PhantomData<ReportedId>,
+}
+impl<ReportedId> DeriveAndReportId<(), ReportedId> {
+	pub fn auto() -> AutoId<ReportedId> {
+		Self { params: (), _phantom: PhantomData }
+	}
+}
+impl<Params, ReportedId> DeriveAndReportId<Params, ReportedId> {
+	pub fn from(params: Params) -> Self {
+		Self { params, _phantom: PhantomData }
+	}
+}
+impl<Params, ReportedId> IdAssignment for DeriveAndReportId<Params, ReportedId> {
+	type ReportedId = ReportedId;
+}
+
+/// Represents the value of an [InspectStrategy] to be used as a configuration value in the
+/// [WithConfig] strategy.
+#[derive(Debug, PartialEq, Eq, Clone, Encode, Decode, MaxEncodedLen, TypeInfo)]
+pub struct ConfigValue<Inspect: InspectStrategy>(pub Inspect::Value);
+
+/// This trait marks a config value to be used in the [WithConfig] strategy.
+/// It is used to make compiler error messages clearer if invalid type is supplied into the
+/// [WithConfig].
+pub trait IsConfigValue {}
```

Comment:
```suggestion
pub trait ConfigValueKind {}
```
IsConfigValue is a misleading name, as it hints towards a boolean value

Reply:
Done, as we spoke in tg